tests/parsenum/main.c

```diff
@@ -5,8 +5,11 @@
 #include "warnp.h"
 
 #define PARSENUM_CMP(x, y)						\
-	(fabs((double)x - (double)y) < 1 &&				\
-	    fabs((double)x - (double)y) <= fabs((double)x + (double)y) * 1e-6)
+	((isnan(x) == isnan(y) && isnan(x) != 0) ||			\
+	    (isinf(x) == isinf(y) && isinf(x) != 0) ||		\
```

Comment:
err... so
```
isinf((double)x) && isinf((double)y) && ((x > 0) && (y > 0) || (x < 0 && y <0))
```
?  I think my original version is simpler.

[I did change the `isnan()` version, though.]